socorro/unittest/cron/jobs/test_update_signatures.py

```diff
@@ -0,0 +1,257 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import mock
+
+from socorro.cron.crontabber_app import CronTabberApp
+from socorro.unittest.cron.jobs.base import IntegrationTestBase
+
+
+class FakeModel(object):
+    def __init__(self):
+        self._get_steps = []
+
+    def add_get_step(self, assertions, response):
+        assertions = assertions or []
+        self._get_steps.append({
+            'assertions': assertions,
+            'response': response
+        })
+
+    def get(self, *args, **kwargs):
+        if not self._get_steps:
+            raise Exception('Unexpected call to .get()')
+
+        step = self._get_steps.pop(0)
+
+        for assertion in step['assertions']:
+            assertion(args, kwargs)
+        return step['response']
+
+
+class UpdateSignaturesCronAppTestCase(IntegrationTestBase):
+    def _setup_config_manager(self):
+        return super(UpdateSignaturesCronAppTestCase, self)._setup_config_manager(
+            jobs_string='socorro.cron.jobs.update_signatures.UpdateSignaturesCronApp|1h'
+        )
+
+    def _truncate_signatures(self):
+        """Wipe the signatures table"""
+        self.conn.cursor().execute('TRUNCATE signatures CASCADE')
+        self.conn.commit()
+
+    def fetch_signatures_data(self):
+        cursor = self.conn.cursor()
+        cursor.execute("SELECT signature, first_build, first_report FROM signatures")
+        results = cursor.fetchall()
+        return [
+            {
+                'signature': str(result[0]),
+                'first_build': str(result[1]),
+                'first_report': str(result[2]),
+            }
+            for result in results
+        ]
+
+    def setUp(self):
+        super(UpdateSignaturesCronAppTestCase, self).setUp()
+        self._truncate_signatures()
+
+    def tearDown(self):
+        super(UpdateSignaturesCronAppTestCase, self).tearDown()
+        self._truncate_signatures()
+
+    @mock.patch('socorro.cron.jobs.update_signatures.SuperSearch')
+    def test_no_crashes_to_process(self, mock_supersearch):
+        supersearch = FakeModel()
+        mock_supersearch.return_value = supersearch
+
+        # Mock SuperSearch to return no results
+        supersearch.add_get_step(
+            assertions=None,
+            response={
+                'errors': [],
+                'hits': [],
+                'total': 0,
+                'facets': {}
+            }
+        )
+
+        # Run crontabber
```

Comment:
There's a lot of repeated steps in these tests, could this be parameterized into a single test?